Source/WebCore/rendering/svg/RenderSVGResourceLinearGradient.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2006 Nikolas Zimmermann <zimmermann@kde.org>
+ * Copyright (C) Research In Motion Limited 2010. All rights reserved.
+ * Copyright (C) 2021, 2022, 2023 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "RenderSVGResourceLinearGradient.h"
+
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+#include "RenderSVGResourceLinearGradientInlines.h"
+#include <wtf/IsoMallocInlines.h>
+
+namespace WebCore {
+
+WTF_MAKE_ISO_ALLOCATED_IMPL(RenderSVGResourceLinearGradient);
+
+RenderSVGResourceLinearGradient::RenderSVGResourceLinearGradient(SVGLinearGradientElement& element, RenderStyle&& style)
+    : RenderSVGResourceGradient(Type::SVGResourceLinearGradient, element, WTFMove(style))
+{
+}
+
+RenderSVGResourceLinearGradient::~RenderSVGResourceLinearGradient() = default;
+
+void RenderSVGResourceLinearGradient::collectGradientAttributesIfNeeded()
+{
+    if (m_attributes.has_value())
+        return;
+
+    m_attributes = std::nullopt;
+    linearGradientElement().synchronizeAllAttributes();
+
+    auto attributes = LinearGradientAttributes { };
+    if (linearGradientElement().collectGradientAttributes(attributes))
+        m_attributes = WTFMove(attributes);
+}
+
+RefPtr<Gradient> RenderSVGResourceLinearGradient::createGradient(const RenderStyle& style)
+{
+    if (!m_attributes)
+        return nullptr;
+
+    auto startPoint = SVGLengthContext::resolvePoint(&linearGradientElement(), m_attributes->gradientUnits(), m_attributes->x1(), m_attributes->y1());
+    auto endPoint = SVGLengthContext::resolvePoint(&linearGradientElement(), m_attributes->gradientUnits(), m_attributes->x2(), m_attributes->y2());
+
+    return Gradient::create(
+        Gradient::LinearData { startPoint, endPoint },
+        { ColorInterpolationMethod::SRGB { }, AlphaPremultiplication::Unpremultiplied },
```

Comment:
I don't believe I ever ended up doing this in the SVG code, but if you are looking at this, one nice improvement you could look at making would be to support the "color-interpolation" property (https://www.w3.org/TR/SVG2/painting.html#ColorInterpolationProperty) and allow users to use linearSRGB here if they wanted. (Same with radial gradients).

The platform Gradient code knows how draw them, we just never piped it in).

(obviously, not for this PR, just wanted to point it out)

Reply:
Great idea, thanks @weinig for the reminder. We should open a ticket to track it.